src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts

```diff
@@ -153,16 +153,28 @@ export class AslVisualization {
         )
 
         // When the panel is closed, dispose of any disposables/remove subscriptions
-        panel.onDidDispose(() => {
+        const disposePanel = () => {
             debouncedUpdate.cancel()
             this.onVisualizationDisposeEmitter.fire()
             this.disposables.forEach(disposable => {
                 disposable.dispose()
             })
             this.onVisualizationDisposeEmitter.dispose()
             this.isPanelDisposed = true
```

Comment:
Consider setting the disposed flag first to prevent the chance of re-entering this function